hw/ip/otbn/doc/otbn_intro.md

```diff
@@ -0,0 +1,390 @@
+# Introduction to OTBN
+
+OTBN (the **O**pen**T**itan **B**ig **N**umber accelerator) is a specialized coprocessor designed for cryptography.
+It runs as part of OpenTitan in addition to the main processor, Ibex.
+The OTBN hardware block could also run as part of a different system and interact with a different main processor, but this page will focus on the OpenTitan context.
+
+This page is an introduction and overview of OTBN.
+For more detailed information, see the [technical specification](../README.md) and the [ISA guide](isa.md).
+
+## How OTBN executes programs
+
+The following diagram illustrates how OTBN and the main processor interact:
+
+![Diagram showing OTBN and Ibex interaction](otbn_operation.svg)
+
+OTBN has its own data and instruction memories, distinct from the memories of the Ibex main processor.
+To run an OTBN program, Ibex loads the program into OTBN's instruction memory (and data memory if the program contains stored constants), then writes the input data into the data memory and flips the "execute" bit.
+When OTBN is done, it sends an interrupt to Ibex, which reads back the results from data memory.
+See [below](#an-example-program) for an example of a standalone OTBN program.
+
+OTBN protects its data from Ibex in various ways:
```

Comment:
I think an extra bullet here mentioning Key Sideloading from the isolated keymgr connection as part of the security model would be helpful.